starlark/int.go

```diff
@@ -26,7 +26,15 @@ func MakeInt64(x int64) Int {
 		}
 		return Int{&smallint[x]}
 	}
-	return Int{new(big.Int).SetInt64(x)}
+	if 0 <= x && int64(int32(x)) == x {
```

Comment:
Let's factor the 0 <= x condition with the one on L23 using a nested if.

The condition checks that the number is a nonnegative int32, but the optimization is valid even for larger values if big.Word is 64 bits. So let's say int64(big.Word(x)) == x.

Reply:
Done, thanks. Also updated comment and applied same optimization to MakeUint64 (oops).